vars/influxWriteData.groovy

```diff
@@ -76,15 +85,18 @@ private void writeToInflux(config, script){
             $class: 'InfluxDbPublisher',
             selectedTarget: config.influxServer,
             customPrefix: config.influxPrefix,
-            customData: script.commonPipelineEnvironment.getInfluxCustomData(),
-            customDataMap: script.commonPipelineEnvironment.getInfluxCustomDataMap()
+            customData: config.customData.size()>0 ? config.customData : null,
+            customDataTags: config.customDataTags.size()>0 ? config.customDataTags : null,
+            customDataMap: config.customDataMap.size()>0 ? config.customDataMap : null,
+            customDataMapTags: config.customDataMapTags.size()>0 ? config.customDataMapTags : null
         ])
     }
 
     //write results into json file for archiving - also benefitial when no InfluxDB is available yet
     def jsonUtils = new JsonUtils()
-    writeFile file: 'jenkins_data.json', text: jsonUtils.getPrettyJsonString(script.commonPipelineEnvironment.getInfluxCustomData())
-    writeFile file: 'pipeline_data.json', text: jsonUtils.getPrettyJsonString(script.commonPipelineEnvironment.getInfluxCustomDataMap())
+    writeFile file: 'jenkins_data.json', text: jsonUtils.getPrettyJsonString(config.customData)
+    writeFile file: 'pipeline_data.json', text: jsonUtils.getPrettyJsonString(config.customDataMap)
+    writeFile file: 'jenkins_data_tags.json', text: jsonUtils.getPrettyJsonString(config.customDataTags)
+    writeFile file: 'pipeline_data_tags.json', text: jsonUtils.getPrettyJsonString(config.customDataMapTags)
```

Comment:
should be named `'influx_data_tag.json'` as it contains `pipeline_data` and `step.data`

Reply:
done